server/services/store/mattermostauthlayer/mattermostauthlayer.go

```diff
@@ -910,7 +910,13 @@ func (s *MattermostAuthLayer) GetBoardsForUserAndTeam(userID, teamID string, inc
 	}
 
 	boards, err := s.Store.GetBoardsInTeamByIds(boardIDs, teamID)
-	if err != nil {
+	// ToDo: check if the query is being used appropriately from the
+	//       interface, Wiggin77 thinks we're passing IDs that belong
```

Comment:
I tried removing the filter on templates from the store method and got the same mismatched lengths, so I'm not so sure templates are the issue.  Why do we filter out templates from the query anyway?  If a template id is in the list why not return it?

Reply:
I don't see a reason to filter out templates in this method, and the `mmauthlayer` method is the only caller, so I'm removing the filter

Reply:
Fixed on [146f4c6](https://github.com/mattermost/focalboard/pull/3842/commits/146f4c67b207a29b45a2b91488645fda9105ed23)